src/main/kotlin/reminder/application/ReminderDeleter.kt

```diff
@@ -0,0 +1,17 @@
+package reminder.application
+
+import reminder.domain.ReminderIdentifier
+import reminder.domain.Reminder
+import reminder.domain.ReminderRepository
+
+class ReminderDeleter(
+    private val repository: ReminderRepository,
+) {
+
+    /***
+     * Deletes a Reminder with the given identifier and returns true if found.
+     * @see Reminder
+     * @return A boolean indicating if the reminder was deleted correctly.
+     */
+    fun delete(identifier: ReminderIdentifier) = repository.delete(identifier)
```

Comment:
As stated on #4, the use case should:
1) Receive the reminder as a parameter.
2) Search if the reminder exists on the database ((you can use the already done ReminderSearcher use case (see
Use case of searching for reminders #3) )
3) If it doesn't exist throw an exception.
4) Remove the reminder.

The exceptions that the function throws should be stated with the "@throws" keyword in the kdoc (kotlin alternative to javadoc).

Reply:
Why should it receive the reminder as a parameter, shouldn't it instead receive just the identifier?

Reply:
Yes, my bad, it should only receive the ReminderIdentifier, but the point I wanted to emphasize was the exception if the reminder doesn't exist.